src/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -86,7 +86,11 @@ Tensor Schedule::cache_read(const Tensor& tensor,
       return tensor(Array<Expr>(i.begin(), i.end()));
     }, os.str());
   std::unordered_map<Tensor, Tensor> vsub;
-  vsub[tensor] = cache;
+  Tensor sugar_tensor = tensor;
+  Stage s = operator[](tensor->op);
+  if (!(s->op.same_as(tensor->op)))
+      sugar_tensor = s->op.output(0);
```

Comment:
Use the sugar tensor’s value index instead of 0 when pass to output

Reply:
Add {} to multiple line if

Reply:
it seems very strange to blocked at 'test_topi_reduce.test_reduce_map', the only different  between "Always use new tensor in its stage when rewrite for cache read #748" is that , the using of sugar tensor is more rigorous, so my change seems be more similar to the origin code . But #748 has passed the test.